File: basic/recursion/reverse_array.cpp
#include <iostream>
using namespace std;
void reverse(vector <int>& arr,int start,int end){
    if(start >= end) return;
    swap(arr[start],arr[end]);
    reverse(arr,start+=1,end-=1);
}
int main(){
    vector <int> arr = {3,4,6,7,8,9,34};
    reverse(arr,0,arr.size()-1);
    for(int i=0; i<arr.size(); i++){
        cout << arr[i] << " ";
    }cout << endl;
}

File: Basic/Recursion/reverse_array.java
class reverse_array{
    static void helper(int arr[],int start,int end){
        if(start >= end) return;
        int temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;
        helper(arr,start+1,end-1);
    }
    static void reverseArray(int arr[]) {
        helper(arr,0,arr.length-1);
    }
    public static void main(String[] args) {
        int[] arr = {5,6,7,8,8,3};
        reverseArray(arr);
    }
}